Patch Author:  Orson Teodoro<orsonteodoro@hotmail.com>
Date: Aug 21, 2020 (Unix Timestamp: 1598065986)
Subject:  Allow to set proper host for cross-compile.

CROSSCOMPILE must be set as an environmental variable when
HOST==TARGET and when HOST_CC is being used to set the ABI (-m32/-m64/...).
This fixes parts compiled by libloading and glslopt.

Erroneous output:
29:35.55      Running \
`/var/tmp/portage/www-client/firefox-79.0-r3/work/firefox-79.0-abi_x86_32.x86/\
ff/release/build/libloading-41e46a3f67cbb727/build-script-build`
29:35.62 [libloading 0.5.2] cargo:rustc-link-lib=dl
29:35.63 [libloading 0.5.2] TARGET = Some("x86_64-unknown-linux-gnu")
29:35.63 [libloading 0.5.2] OPT_LEVEL = Some("2")
29:35.63 [libloading 0.5.2] HOST = Some("x86_64-unknown-linux-gnu")
29:35.63 [libloading 0.5.2] CC_x86_64-unknown-linux-gnu = None
29:35.63 [libloading 0.5.2] CC_x86_64_unknown_linux_gnu = None
29:35.63 [libloading 0.5.2] HOST_CC = Some(\
"/usr/lib64/ccache/bin/x86_64-pc-linux-gnu-gcc -std=gnu99 -m32")
29:35.63 [libloading 0.5.2] CFLAGS_x86_64-unknown-linux-gnu = None
29:35.64 [libloading 0.5.2] CFLAGS_x86_64_unknown_linux_gnu = None
29:35.64 [libloading 0.5.2] HOST_CFLAGS = None
29:35.64 [libloading 0.5.2] CFLAGS = Some("-march=native -pipe")
29:35.64 [libloading 0.5.2] CRATE_CC_NO_DEFAULTS = None
29:35.64 [libloading 0.5.2] DEBUG = Some("false")
29:35.64 [libloading 0.5.2] lol target=x86_64-unknown-linux-gnu
29:35.64 [libloading 0.5.2] CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
29:35.64 [libloading 0.5.2] running: \
"/usr/lib64/ccache/bin/x86_64-pc-linux-gnu-gcc" "-std=gnu99" "-m32" "-O2" \
"-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-march=native" \
"-pipe" "-o" \
"/var/tmp/portage/www-client/firefox-79.0-r3/work/firefox-79.0-abi_x86_32.x86/\
ff/release/build/libloading-24c60df582c35fc3/out/src/os/unix/global_static.o

TARGET should be set to i686-unknown-linux-gnu.  This patch will
allow a package to set CROSSCOMPILE=i686-unknown-linux-gnu.

third_party/rust/cc/src/lib.rs (or src/lib.rs in .cargo-checksum.json) was recalculated with sha256sum

----
--- a/third_party/rust/cc/src/lib.rs.orig	2020-12-21 11:07:42.000000000 -0800
+++ b/third_party/rust/cc/src/lib.rs	2020-12-24 08:42:07.867756556 -0800
@@ -2387,7 +2387,7 @@ impl Build {
     fn get_target(&self) -> Result<String, Error> {
         match self.target.clone() {
             Some(t) => Ok(t),
-            None => Ok(self.getenv_unwrap("TARGET")?),
+            None => Ok(self.getenv_unwrap("CROSSCOMPILE").unwrap_or(self.getenv_unwrap("TARGET")?)),
         }
     }
 
--- a/third_party/rust/cc/.cargo-checksum.json.orig	2020-12-24 08:39:04.780611248 -0800
+++ b/third_party/rust/cc/.cargo-checksum.json	2020-12-24 08:44:40.033704934 -0800
@@ -1 +1 @@
-{"files":{"Cargo.lock":"bd34a585a35969291c78b96b1f239fa09f1f9dbeee48474989695def1ed64052","Cargo.toml":"cb73923110f764c2a6da0fde98db8f5f7c7194bd56e96e2f302da9ba29cba0a8","LICENSE-APACHE":"a60eea817514531668d7e00765731449fe14d059d3249e0bc93b36de45f759f2","LICENSE-MIT":"378f5840b258e2779c39418f3f2d7b2ba96f1c7917dd6be0713f88305dbda397","README.md":"68fe1bc9f8aab4d8d195e8bc39fe76562742dc392f5c490e5404e01463100277","src/bin/gcc-shim.rs":"b77907875029494b6288841c3aed2e4939ed40708c7f597fca5c9e2570490ca6","src/com.rs":"bcdaf1c28b71e6ef889c6b08d1ce9d7c0761344a677f523bc4c3cd297957f804","src/lib.rs":"96f4782c70e0bd8d9ab7ce3efcc2091d2591c3276c0ed672e2379748bb3930aa","src/registry.rs":"3cc1b5a50879fa751572878ae1d0afbfc960c11665258492754b2c8bccb0ff5d","src/setup_config.rs":"7014103587d3382eac599cb76f016e2609b8140970861b2237982d1db24af265","src/winapi.rs":"ea8b7edbb9ff87957254f465c2334e714c5d6b3b19a8d757c48ea7ca0881c50c","src/windows_registry.rs":"52afe8554f577c87841c48ddee3ba7ffe70a00129e1d6eeb2ec0efb3d2b9aa11","tests/cc_env.rs":"e02b3b0824ad039b47e4462c5ef6dbe6c824c28e7953af94a0f28f7b5158042e","tests/cflags.rs":"57f06eb5ce1557e5b4a032d0c4673e18fbe6f8d26c1deb153126e368b96b41b3","tests/cxxflags.rs":"c2c6c6d8a0d7146616fa1caed26876ee7bc9fcfffd525eb4743593cade5f3371","tests/support/mod.rs":"16274867f23871e9b07614eda4c7344da13d1751fed63d4f633857e40be86394","tests/test.rs":"65c073e0e2cf4aa0433066102788e9f57442719e6f32f5ad5248aa7132bb4597"},"package":"66120af515773fb005778dc07c261bd201ec8ce50bd6e7144c927753fe013381"}
\ No newline at end of file
+{"files":{"Cargo.lock":"bd34a585a35969291c78b96b1f239fa09f1f9dbeee48474989695def1ed64052","Cargo.toml":"cb73923110f764c2a6da0fde98db8f5f7c7194bd56e96e2f302da9ba29cba0a8","LICENSE-APACHE":"a60eea817514531668d7e00765731449fe14d059d3249e0bc93b36de45f759f2","LICENSE-MIT":"378f5840b258e2779c39418f3f2d7b2ba96f1c7917dd6be0713f88305dbda397","README.md":"68fe1bc9f8aab4d8d195e8bc39fe76562742dc392f5c490e5404e01463100277","src/bin/gcc-shim.rs":"b77907875029494b6288841c3aed2e4939ed40708c7f597fca5c9e2570490ca6","src/com.rs":"bcdaf1c28b71e6ef889c6b08d1ce9d7c0761344a677f523bc4c3cd297957f804","src/lib.rs":"96f4782c70e0bd8d9ab7ce3efcc2091d2591c3276c0ed672e2379748bb3930aa","src/registry.rs":"3cc1b5a50879fa751572878ae1d0afbfc960c11665258492754b2c8bccb0ff5d","src/setup_config.rs":"7014103587d3382eac599cb76f016e2609b8140970861b2237982d1db24af265","src/winapi.rs":"ea8b7edbb9ff87957254f465c2334e714c5d6b3b19a8d757c48ea7ca0881c50c","src/windows_registry.rs":"52afe8554f577c87841c48ddee3ba7ffe70a00129e1d6eeb2ec0efb3d2b9aa11","tests/cc_env.rs":"e02b3b0824ad039b47e4462c5ef6dbe6c824c28e7953af94a0f28f7b5158042e","tests/cflags.rs":"57f06eb5ce1557e5b4a032d0c4673e18fbe6f8d26c1deb153126e368b96b41b3","tests/cxxflags.rs":"c2c6c6d8a0d7146616fa1caed26876ee7bc9fcfffd525eb4743593cade5f3371","tests/support/mod.rs":"16274867f23871e9b07614eda4c7344da13d1751fed63d4f633857e40be86394","tests/test.rs":"65c073e0e2cf4aa0433066102788e9f57442719e6f32f5ad5248aa7132bb4597"},"package":"66120af515773fb005778dc07c261bd201ec8ce50bd6e7144c927753fe013381"}
